src/element/writer.rs

```diff
@@ -54,3 +92,30 @@ pub enum Format {
     Binary,
     // TODO a mode for Json(TextKind)
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::element::owned::Element;
+    use crate::element::writer::ElementWriter;
+    use crate::ion_eq::IonEq;
+    use crate::text::text_writer::TextWriterBuilder;
+
+    use crate::{IonResult, IonWriter};
+    use nom::AsBytes;
+
+    #[test]
+    fn element_roundtrip() -> IonResult<()> {
+        let mut buffer = Vec::new();
+        let mut writer = TextWriterBuilder::new().build(&mut buffer)?;
+
+        let ion = r#"
+            null true 0 1e0 2.0 2022T foo "bar" (foo bar baz) [foo, bar, baz] {foo: true, bar: false}
+        "#;
+        let expected_elements = Element::read_all(ion.as_bytes())?;
```

Comment:
Would the semantics of the test be changed by using a builder macro for `expected_elements`? I feel like that would be an improvement but perhaps I'm missing something.

Separately, this is read text, write text, read text. Is there a version of this that round trips binary or round trips across formats, e.g. read text, write binary, then read binary?